iocore/cache/Cache.cc

```diff
@@ -1041,12 +1041,12 @@ CacheProcessor::cacheInitialized()
       case CACHE_COMPRESSION_FASTLZ:
         break;
       case CACHE_COMPRESSION_LIBZ:
-#if !TS_HAS_LIBZ
+#ifndef HAVE_ZLIB_H
```

Comment:
So, while I'm generally OK with this (I think), I'm slightly concerned that we don't have a universal pattern for the #if vs #ifdef. In general, I think we ought to prefer #if over $ifdef's, assuming it's defined to "1". This helps avoid the problem where #define foo 0  is actually "defined", but is off.

I know that we have plenty of cases where we do #ifndef / #ifdef, but unless there are other reasons to use that here, I think we should stick to #if when possible for these boolean types. Ideally, we'd later change this to be consistent as well.